libnavigation-util/src/test/java/com/mapbox/navigation/utils/timer/MapboxTimerTest.kt

```diff
@@ -1,94 +1,114 @@
 package com.mapbox.navigation.utils.timer
 
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.delay
-import kotlinx.coroutines.runBlocking
-import kotlinx.coroutines.test.runBlockingTest
-import kotlinx.coroutines.test.setMain
-import org.junit.Assert.assertEquals
-import org.junit.Assert.assertFalse
-import org.junit.Before
+import com.mapbox.navigation.testing.MainCoroutineRule
+import io.mockk.Called
+import io.mockk.coVerify
+import io.mockk.mockk
+import java.util.concurrent.TimeUnit
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancelAndJoin
+import org.junit.Rule
 import org.junit.Test
 
+@ExperimentalCoroutinesApi
 class MapboxTimerTest {
 
-    @Before
-    fun setUp() {
-        Dispatchers.setMain(Dispatchers.Default)
-    }
+    @get:Rule
+    val coroutineRule = MainCoroutineRule()
 
-    @Test
-    fun start() = runBlocking {
-        var counter = 0
-        val testLambda = { counter += 1 }
+    private val mapboxTimer = MapboxTimer()
+    private val mockLambda: () -> Unit = mockk(relaxed = true)
 
-        MapboxTimer(100L, testLambda).start()
-        delay(220)
+    @Test
+    fun `should not call before interval`() = coroutineRule.runBlockingTest {
```

Comment:
NIT Normally is good to structure the tests following the triple A 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=40 http://wiki.c2.com/?ArrangeActAssert
If the `A`rrange part gets too long we should create factory or builder methods.

The same thing applies to the rest of the tests 👇

Reply:
updated the whitespace, is more AAA now. Arrange is setting the timer interval, so only a single line